### PR TITLE
Coinpage

### PR DESCRIPTION
**Pull request by kabelic**
Fetch from api:
- bitcoin data
- ethereum
- dai

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -71,6 +74,9 @@ class App extends React.Component {
                 <Routes>
                   <Route path='/' element={<Home/>}/>
                   <Route path='/coin' element={<Coin/>}/>
+                  <Route path='/coin/dai' element={<Dai/>}/>
+                  <Route path='/coin/bitcoin' element={<Bitcoin/>}/>
+                  <Route path='/coin/ethereum' element={<Ethereum/>}/>
```
Comment (yoloOnTheBattlefield) [inline src/App.js:79]: you should make the coin page dynamic, checkout how you did the weather app

File: src/components/BitcoinLeft.js [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+                            <BitcoinIcon bgColor={'#272727'}>
+                                { item.name ? <img src={item.image} width='15' alt='Bitcoin'/> :  <BsCurrencyBitcoin/> }
+                            </BitcoinIcon>
+                            <TextStyle margin={10}>
+                                <Header weight={400} size={15}>{item.name ? item.name : 'Bitcoin'}</Header>
+                                <SubHeader weight={300} size={10}>{ item.name ? item.symbol.toUpperCase() : 'BTC' }</SubHeader>
+                            </TextStyle>
+                        </BitcoinLeft1div>
+                    )
+                }
```
Comment (yoloOnTheBattlefield) [inline src/components/BitcoinLeft.js:31]: you need to return null from map if the if statement bounces

File: src/components/Chart.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { BsCurrencyBitcoin } from 'react-icons/bs'
+import { ChartType, ChartTop, ChartTopLeft, ChartTopRight, ChartTopLeftInner, ButtonText, ChartBottom } from "./styles/coinpage.styled"
+import { TextStyle, Header, SubHeader, BitcoinIcon } from "./styles/Body.styled"
+import BarChart from './Charts/BarChart'
+import LineChart from './Charts/LineChart'
+import { UserData } from '../utils/Data'
+
+const Chart = () => {
+    const [userData, setUserData] = useState({
+        labels: UserData.map((data) => data.year),
```
Comment (yoloOnTheBattlefield) [inline src/components/Chart.js:11]: use camelCase

File: src/components/Charts/LineChart.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { Line } from 'react-chartjs-2'
+// import {Chart as ChartJS} from 'chart.js/auto'
```
Comment (yoloOnTheBattlefield) [inline src/components/Charts/LineChart.js:2]: remove commented out code

File: src/components/navComponent/NavComponent.js [hunk 5]
```diff
@@ -2,16 +2,18 @@ import { MdNotifications } from 'react-icons/md'
...
 const NavComponent = (props) => {
     return(
         <Nav>
             <Search/>
-            <CreateAccount bg={'#FFAF2C'} text={'Create a new account'}/>
-            <Notification>
-                <MdNotifications color='#FFAF2C'/>
-            </Notification>
+            <SearchRigh>
+                <CreateAccount bg={'#FFAF2C'} text={'Create a new account'}/>
```
Comment (yoloOnTheBattlefield) [inline src/components/navComponent/NavComponent.js:12]: don't use {} for strings

File: src/components/styles/Body.styled.js [hunk 6]
```diff
@@ -17,7 +17,10 @@ import styled from "styled-components";
    width: 100%;
  `
  export const TextStyle = styled.div`
-   margin-top: ${({ margin }) => margin }px;
+   display: flex;
+   flex-direction: column;
+   justify-content: center;
+   margin-left: ${({ margin }) => margin }px;
```
Comment (yoloOnTheBattlefield) [inline src/components/styles/Body.styled.js:23]: remove empty space

File: src/components/styles/Body.styled.js [hunk 7]
```diff
@@ -144,10 +141,13 @@ export const LogoTitle = styled.div`
...
 `
 export const LogoDescription = styled.div`
+   display: flex;
+   justify-content: center;
    font-family: 'Poppins';
    font-style: normal;
-   font-weight: 500;
-   font-size: 3px;
-   line-height: 4px;
+   /* font-weight: ${(props) => props.size ? props.size : 500}; */
```
Comment (yoloOnTheBattlefield) [inline src/components/styles/Body.styled.js:148]: remove commented out code

File: src/components/styles/Nav.styled.js [hunk 8]
```diff
@@ -116,8 +117,14 @@ export const NewAccount = styled.div`
     justify-content: center;
     background: ${(props) => props.bg ? props.bg : 'transparent'};
     border: 1px solid #FFAF2C;
-    margin: 0 10px;
+    /* margin: 0 10px; */
```
Comment (yoloOnTheBattlefield) [inline src/components/styles/Nav.styled.js:120]: remove commented out code

File: src/components/styles/Nav.styled.js [hunk 9]
```diff
@@ -116,8 +117,14 @@ export const NewAccount = styled.div`
...
+    margin-right: 10px;
     padding: 0 15px ;
     border-radius: 6px;
     color: ${(props) => props.textColor ? props.textColor : 'white'};
+    width: 100%;
+`
```
Comment (yoloOnTheBattlefield) [inline src/components/styles/Nav.styled.js:126]: add new line

File: src/components/styles/coinpage.styled.js [hunk 10]
```diff
@@ -16,24 +17,26 @@ export const NumbeComponentStyle = styled.div`
...
 export const NumberValue = styled.div`
     display: flex;
     font-family: 'Helvetica Now Display';
     font-style: normal;
-    font-weight: 500;
-    font-size: 24px;
-    line-height: 35px;
+    font-weight: 300;
+    font-size: 20px;
+    /* line-height: 35px; */
```
Comment (yoloOnTheBattlefield) [inline src/components/styles/coinpage.styled.js:37]: remove comments

File: src/pages/Ethereum.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+    state={
+        list: null,
+        isLoading: false,
+    }
+
+    getData = async () => {
+        try{
+            this.setState({ isLoading: true })
+            const {data} = await axios('https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d')
```
Comment (yoloOnTheBattlefield) [inline src/pages/Ethereum.js:20]: const {` `data` `}

File: src/pages/Ethereum.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+
+    componentDidMount(){
+        this.getData()
+    }
+
+    render(){
+        return(
+            <div style={{width: '100%'}}>
```
Comment (yoloOnTheBattlefield) [inline src/pages/Ethereum.js:33]: don't use inline styles

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (kabelic) [reply src/App.js:79]: I haven't done weather app before. would you give me a hint?

**Pull request merged**